docs: add debugger ref; remove note from algopy testing doc as its no longer in preview (#368)
algorandfoundation/puya: docs/references/avm_debugger.md

> # AlgoKit AVM VSCode Debugger

Reviewer: ```suggestion
# Breakpoint debugging
```

> The AlgoKit AVM VS Code Debugger extension enables line-by-line debugging of Algorand smart contracts on the AVM, supporting both TEAL and Puya contracts. It provides a seamless debugging experience by leveraging AVM simulate traces and source maps.

Reviewer: ```suggestion
The AlgoKit AVM VS Code Debugger extension enables line-by-line debugging of Algorand Python smart contracts that are executed on the AVM. It provides a seamless debugging experience by leveraging AVM simulate traces and source maps.
```
